test/routing/route_test.cc

```diff
@@ -0,0 +1,580 @@
+    EXPECT_CALL(id_index_, DoGetLane(kLaneIdBA)).WillRepeatedly(Return(&lane_b_a_));
+    EXPECT_CALL(*(road_geometry_ptr_), DoById()).WillRepeatedly(ReturnRef(id_index_));
+    const std::vector<api::RoadPosition> kStartRoadPositionsPhaseA{
+        api::RoadPosition{&lane_a_b_, kStartLanePositionPhaseA}};
+    const std::vector<api::RoadPosition> kEndRoadPositionsPhaseA{api::RoadPosition{&lane_a_a_, kEndLanePositionPhaseA}};
+    const std::vector<api::RoadPosition> kStartRoadPositionsPhaseB{
+        api::RoadPosition{&lane_b_a_, kStartLanePositionPhaseB}};
+    const std::vector<api::RoadPosition> kEndRoadPositionsPhaseB{api::RoadPosition{&lane_b_a_, kEndLanePositionPhaseB}};
+    phase_a_ = std::make_unique<Phase>(kPhaseAIndex, kLaneSRangeTolerance, kStartRoadPositionsPhaseA,
+                                       kEndRoadPositionsPhaseA, kLaneSRangesPhaseA, road_network_.get());
+    phase_b_ = std::make_unique<Phase>(kPhaseBIndex, kLaneSRangeTolerance, kStartRoadPositionsPhaseB,
+                                       kEndRoadPositionsPhaseB, kLaneSRangesPhaseB, road_network_.get());
+  }
+
+  LaneMock lane_a_a_;
+  LaneMock lane_a_b_;
+  LaneMock lane_b_a_;
+  SegmentMock segment_a_;
+  SegmentMock segment_b_;
+  std::unique_ptr<Phase> phase_a_;
+  std::unique_ptr<Phase> phase_b_;
+};
+
+TEST_F(RouteWithTwoPhasesTest, SizeIsTwo) {
+  const Route dut({*phase_a_, *phase_b_}, road_network_.get());
+
+  EXPECT_EQ(2, dut.size());
+}
+
+TEST_F(RouteWithTwoPhasesTest, GetCanIndexPhases) {
+  const Route dut({*phase_a_, *phase_b_}, road_network_.get());
+
+  EXPECT_EQ(kPhaseAIndex, dut.Get(kPhaseAIndex).index());
+  EXPECT_EQ(kPhaseBIndex, dut.Get(kPhaseBIndex).index());
+}
+
+TEST_F(RouteWithTwoPhasesTest, StartRoutePositionIsOnLaneAB) {
+  const Route dut({*phase_a_, *phase_b_}, road_network_.get());
+
+  EXPECT_EQ(&lane_a_b_, dut.start_route_position().lane);
+  EXPECT_EQ(kStartLanePositionPhaseA.srh(), dut.start_route_position().pos.srh());
+}
+
+TEST_F(RouteWithTwoPhasesTest, EndRoutePositionIsOnLaneBA) {
+  const Route dut({*phase_a_, *phase_b_}, road_network_.get());
+
+  EXPECT_EQ(&lane_b_a_, dut.end_route_position().lane);
+  EXPECT_EQ(kEndLanePositionPhaseB.srh(), dut.end_route_position().pos.srh());
+}
+
+TEST_F(RouteWithTwoPhasesTest, FindRoutePositionByInertialPositionIsMappedViaPhaseOntoLaneSRangeAA) {
+  // To simplify the modelled situation, please revisit the following diagram where the o
+  // indicates where the InertialPosition falls.
+  // <pre>
+  //
+  //      Phase A        Phase B
+  //  x-o----AA------*------BA------*
+  //  *------AB------x
+  //
+  // </pre>
+  const api::LanePositionResult kLaneAAPositionResult{api::LanePosition{10., 0., 0.},
+                                                      api::InertialPosition{10., 5., 0.}, 0.};
+  const api::LanePositionResult kLaneABPositionResult{api::LanePosition{10., 5., 0.},
+                                                      api::InertialPosition{10., 5., 0.}, 0.};
```

Comment:
Mmmm. I think it is only correct if you are calling `Lane::ToSegmentPosition`, not when you are calling `ToLanePosition`.

If inertial position is  `api::InertialPosition{10., 5., 0.}` the `LanePositionResult` will be 

```cpp
  const api::LanePositionResult kLaneABPositionResult{api::LanePosition{10., 2.5, 0.},
                                                      api::InertialPosition{10., 2.5, 0.}, 2.5};
```